lib/rialto/etl/service_client/entity_resolver.rb

```diff
@@ -22,7 +22,13 @@ def initialize
 
         def resolve(type, params)
           resp = conn.get(type, params)
-          return resp.body if resp.success?
+          if resp.success?
+            return RDF::URI.new(resp.body) if resp.success?
+          elsif resp.status == 404
+            return nil
+          else
+            raise Error, "Entity resolver returned #{resp.status} for #{type} type and #{params} params."
```

Comment:
Where is `Error` defined? I'm guessing this should be either `StandardError` or `RuntimeError`, but a more specific exception class would work too.

Reply:
Do you have a suggested exception class? I had briefly looked, but couldn't find something I was happy with.

Reply:
Go with `RuntimeError` then.